ajax-project/js/main.js

```diff
@@ -57,35 +59,56 @@ window.addEventListener('load', function () {
   if ($beerWindow.getAttribute('class') !== 'beer-window hidden') {
     getBeerData();
   }
+  if ($comboWindow.getAttribute('class') !== 'taco-and-beer-window hidden') { getTacoRecipe(); }
+  getBeerData();
 });
 
-var $recipe = document.querySelector('#recipe-window');
-
 document.addEventListener('click', function () {
-  if (event.target.getAttribute('id') === 'cycle-recipe') {
+
+  if (event.target.getAttribute('id') === 'cycle-recipe' || event.target.getAttribute('id') === 'cycle-combo-recipe') {
     isLoading = true;
     getTacoRecipe();
   }
-  if (event.target.getAttribute('id') === 'save-recipe') {
+  if (event.target.getAttribute('id') === 'save-recipe' || event.target.getAttribute('id') === 'save-combo-recipe') {
     if (isLoading) return;
     saveTaco();
   }
-  if (event.target.getAttribute('id') === 'cycle-beer') {
+  if (event.target.getAttribute('id') === 'cycle-beer' || event.target.getAttribute('id') === 'cycle-combo-beer') {
     isLoading = true;
     getBeerData();
   }
-  if (event.target.getAttribute('id') === 'save-beer') {
+  if (event.target.getAttribute('id') === 'save-beer' || event.target.getAttribute('id') === 'save-combo-beer') {
     if (isLoading) return;
     saveBeer();
   }
-  if (event.target.getAttribute('id') === 'save-recipe' || event.target.getAttribute('id') === 'save-beer') {
+
+  if (event.target.getAttribute('id') === 'save-recipe' || event.target.getAttribute('id') === 'save-beer' || event.target.getAttribute('id') === 'save-combo' || event.target.getAttribute('id') === 'save-combo-beer' || event.target.getAttribute('id') === 'save-combo-recipe') {
     var dataJson = JSON.stringify(data);
     localStorage.setItem('saved-tacos-and-beer', dataJson);
   }
```

Comment:
This could be must more optimized with a set of if and else if statements

In its current state, even when a condition is met, the computer will try to evalulate all the comparisons you have set.  If they are organized in a if / else if block, then once a condition is met, it won't try to do any more checking.

And also if you stored the id at the top of this function you won't have to call getAttribute a bunch of time

```
var id = event.target.id
```

Reply:
Also come to think of it, aren't all these elements in one container? It would be better to put the event listener on the parent container of the navbar instead of the document, so there isnt this function fired off everytime the document is clicked.